### PR TITLE
Set/Get for multivectors

### PR DESCRIPTION
**Pull request by victorapm**
This PR extends the Set/AddTo/Get routines to work with multivectors. Additionally:

1. Add `HYPRE_IJVectorInnerProd`
2. Add new set of regression tests `TEST_ij/vector`

Regression tests:
- [x] Tux
- [x] Lassen

### CODE REVIEW

**Review by ulrikeyang — APPROVED** [review]
This looks good.

**Review by rfalgout — APPROVED** [review]
Looks good, Victor!  We should check to see if this will affect MFEM since they touch our data structures.

**Comment by victorapm** [comment]
> We should check to see if this will affect MFEM

@rfalgout Thanks! I'll do that.

**Comment by victorapm** [comment]
Hi all,

I built MFEM with this branch, ran `make test`, and all tests passed.

@waynemitchell, I see this interacts with #676, do you prefer any specific merging order? Thanks!

**Comment by waynemitchell** [comment]
> Hi all,
> 
> I built MFEM with this branch, ran `make test`, and all tests passed.
> 
> @waynemitchell, I see this interacts with #676, do you prefer any specific merging order? Thanks!

@victorapm, I'm fine if you want to go ahead and merge first. I'll go ahead and merge your branch into mine and fix conflicts.

File: src/IJ_mv/IJVector_parcsr.c [hunk 1]
```diff
@@ -264,6 +268,42 @@ hypre_IJVectorZeroValuesPar(hypre_IJVector *vector)
...
+   hypre_ParVector *par_vector  = (hypre_ParVector*) hypre_IJVectorObject(vector);
+   HYPRE_Int        num_vectors = hypre_ParVectorNumVectors(par_vector);
+
+   if (component < 0 || component > num_vectors)
+   {
+      if (print_level)
+      {
+         hypre_printf("component < 0 || component > num_vectors -- ");
+         hypre_printf("hypre_IJVectorSetComponentPar\n");
+         hypre_printf("**** This vector partitioning should not occur ****\n");
```
Comment (liruipeng) [inline src/IJ_mv/IJVector_parcsr.c]: this message about 'partitioning' doesn't seem to be relevant

**Comment by liruipeng** [comment]
@victorapm This looks good to me in general. But I still want to confirm some things with `stack_voff` in IJVector. Maybe a quick chat is good to have. Thanks!

**Review by liruipeng — APPROVED** [review]
Looks good to me. Thanks!

File: src/test/ij.c [hunk 2]
```diff
@@ -3419,6 +3482,71 @@ main( hypre_int argc,
       x = (HYPRE_ParVector) object;
    }
 
+   /* Compute RHS squared norm */
+   if (ij_b)
+   {
+      HYPRE_IJVectorInnerProd(ij_b, ij_b, &b_dot_b);
```
Comment (waynemitchell) [inline src/test/ij.c]: Hi @victorapm. I'm getting an issue at this line when the RHS is read in from file. The underlying code here tries to do the inner product on the device, but the data is still on the host, which is throwing an error for me.

Comment (victorapm) [reply src/test/ij.c]: Hi Wayne, does it happen when using `-rhsfromfile`?

Comment (victorapm) [reply src/test/ij.c]: Wayne, it should work now. Let me know if you have problems :)

File: src/IJ_mv/IJVector_parcsr.c (hunk 1 above)
Comment (victorapm) [reply src/IJ_mv/IJVector_parcsr.c]: Removed. Thanks!

File: src/test/ij.c (hunk 2 above)
Comment (waynemitchell) [reply src/test/ij.c]: Yes, it was with `-rhsfromfile` and is fixed now. Thanks!

**Comment by victorapm** [comment]
I reran Tux and Lassen tests and they passed, so I'm merging this into master. Thanks!

**Pull request merged**